src/azure-cli/azure/cli/command_modules/backup/custom.py

```diff
@@ -112,14 +112,26 @@
 # pylint: disable=too-many-function-args
 
 
-def create_vault(client, vault_name, resource_group_name, location, tags=None, classic_alerts='Enable',
-                 azure_monitor_alerts_for_job_failures='Enable'):
+def create_vault(client, vault_name, resource_group_name, location, tags=None, public_network_access=None,
+                 classic_alerts='Enable', azure_monitor_alerts_for_job_failures='Enable'):
     vault_sku = Sku(name=SkuName.standard)
+    if public_network_access is None:
+        # get the existing value of public_network_access so the request is made correctly
+        try:
+            existing_vault_if_any = client.get(resource_group_name, vault_name)
+            existing_vault_public_network_access = existing_vault_if_any.properties.public_network_access
+            # TODO add better validation for existing_vault_public_network_access? It might be invalid.
+            #   Maybe have a list of possible values and iterate through, and if not add a warning to
+            #   contact support? Such as: if public_network_access in [list], <action>, else <warn user>.
```

Comment:
@hiaga Can you check this if possible?

Reply:
is the code same for create and update vault in CLI ? why do you need to check here if the vault already exists ?

currently the allowed values are Enabled, Disabled

Reply:
Yep, the code for update and create are handled by the same call.

Curious about the (unlikely?) situation for where a vault might _somehow_ have some other value than what's currently allowed. In that case, do we want to add a check here? Or we can operate on the assumption that the situation is unlikely, and if it is then we don't need to handle it as it already exists and this code won't be *introducing* the bug, just continuing it (as this is for when the vault already exists and has said value).

Reply:
as this is create and update flow, we only care what user is allowed to pass while creation/updation. So we restrict the user to only be able to pass Enable/Disable else we throw an error saying invalid input. 

now the case where an invalid value exist from service end. we can choose to handle or not handle it, It's upto us, as per service currently allowed values are E, D.